### PR TITLE
Add range selection scene.

### PR DESCRIPTION
**Pull request by Ivan-Skoryk**
## Summary
Added range selection scene.

 #### Checklist:
 - [x] I have added a brief description of the problem in the PR body.
 - [x] I have explained how I solved the problem in the PR body.
 - [x] I have covered possible edge cases and side effects.
 - [x] I have have attached screenshots to the PR of any UI change.
 - [x] I have checked that my changes work on iOS and macOS targets.
 - [x] I have checked that my changes work correctly on devices with and without a Home button.
 - [x] I have added unit tests for the new logic.
 - [x] I have achieved maximum possible code coverage for the new code.
 - [x] I have checked that my UI works correctly on both Light and Dark themes.

 #### Acceptance Criteria

 #### Screenshots or GIFs (if appropriate)
Screenshot 2020-05-25 at 20 54 55
Screenshot 2020-05-25 at 20 51 10
Screenshot 2020-05-26 at 12 23 17

### CODE REVIEW

File: xDrip/Resources/en.lproj/Localizable.strings [hunk 1]
```diff
@@ -217,6 +217,17 @@
 "settings_transmitter_reset_unsupported_alert_title" = "Can't proceed";
 "settings_transmitter_reset_unsupported_alert_message" = "Resetting transmitter with your firmware version is unsupported";
 
+// Settings Range Selection
+"settings_range_selection_scene_title" = "Range Selection";
+"settings_range_selection_not_higher_less" = "Not Higher/Less";
+"settings_range_selection_normal_section_header" = "INDICATORS ARE NORMAL";
+"settings_range_selection_high_low" = "High/Less";
```
Comment (Artem-Kalmykov) [inline xDrip/Resources/en.lproj/Localizable.strings:224]: Change value to `High/Low`

File: xDrip/Resources/en.lproj/Localizable.strings [hunk 2]
```diff
@@ -217,6 +217,17 @@
...
+"setttings_range_selection_above_normal_section_header" = "ABOVE/BELOW NORMAL";
+"settings_range_selection_urgent_high_low" = "Urgent High/Less";
```
Comment (Artem-Kalmykov) [inline xDrip/Resources/en.lproj/Localizable.strings:226]: Change value to `Urgent High/Low`

File: xDrip/Resources/en.lproj/Localizable.strings [hunk 3]
```diff
@@ -217,6 +217,17 @@
...
+"settings_range_selection_urgent_section_header" = "CRITICAL LEVELS";
+"settings_range_selection_mgdl" = "mg/dL";
```
Comment (Artem-Kalmykov) [inline xDrip/Resources/en.lproj/Localizable.strings:228]: Reuse global unit strings.

File: xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesInteractor.swift [hunk 4]
```diff
@@ -26,7 +26,38 @@ final class SettingsChartRangesInteractor: SettingsChartRangesBusinessLogic, Set
...
     }
+    
+    private func handlePickerValueChanged(_ field: SettingsChartRanges.Field, _ values: [Double]) {
+        let settings = User.current.settings
+        let unit = settings?.unit ?? .default
+        let step = unit.pickerStep
+        let convertedValues = values.map { GlucoseUnit.convertToDefault($0) }
+        switch field {
+        case .notHigherLess:
+            settings?.configureWarningLevel(.high, value: convertedValues[0])
```
Comment (Artem-Kalmykov) [inline xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesInteractor.swift:40]: Dangerous, add a `guard values.count == 2` at the end of the function.

File: xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesInteractor.swift [hunk 5]
```diff
@@ -26,7 +26,38 @@ final class SettingsChartRangesInteractor: SettingsChartRangesBusinessLogic, Set
...
+                settings?.configureWarningLevel(.urgentLow, value: value)
+            }
+        case .urgent:
+            settings?.configureWarningLevel(.urgentHigh, value: convertedValues[0])
+            settings?.configureWarningLevel(.urgentLow, value: convertedValues[1])
+        default:
+            break
+        }
+        
+        doLoad(request: SettingsChartRanges.Load.Request())
```
Comment (Artem-Kalmykov) [inline xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesInteractor.swift:61]: Rename `Load` use case to `UpdateData`, if you're reusing it multiple times.

File: xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift [hunk 6]
```diff
@@ -19,10 +19,195 @@ protocol SettingsChartRangesPresentationLogic {
 final class SettingsChartRangesPresenter: SettingsChartRangesPresentationLogic {
     weak var viewController: SettingsChartRangesDisplayLogic?
     
+    private lazy var settings: Settings = {
```
Comment (Artem-Kalmykov) [inline xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift:22]: Presenter shouldn't access models directly, they should be passed in `Response` by interactor.

File: xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift [hunk 7]
```diff
@@ -19,10 +19,195 @@ protocol SettingsChartRangesPresentationLogic {
...
+                .notHigherLess,
+                detailValues: [highValue, lowValue],
+                pickerValueChanged: response.pickerValueChanged
+            )
+        ]
+        
+        let attrHeader = NSMutableAttributedString()
+        
+        let textAttachment = NSTextAttachment()
+        textAttachment.image = UIImage(named: "greenOval")
```
Comment (Artem-Kalmykov) [inline xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift:58]: Wouldn't it be more convenient to not use `AttributtedString` for this, but pass a custom `UIView` instead?

File: xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift [hunk 8]
```diff
@@ -19,10 +19,195 @@ protocol SettingsChartRangesPresentationLogic {
...
+        return .normal(cells: cells, header: nil, footer: nil, attributedHeader: attrHeader, attributedFooter: nil)
+    }
+    
+    private func createPickerCell(
+        _ field: SettingsChartRanges.Field,
+        detailValues: [Double],
+        pickerValueChanged: @escaping (SettingsChartRanges.Field, [Double]) -> Void) -> BaseSettings.Cell {
+        let range = settings.unit.minMax
+        let step = settings.unit.pickerStep
+        var array1 = [Double]()
```
Comment (Artem-Kalmykov) [inline xDrip/Scenes/Settings/SettingsChartRanges/SettingsChartRangesPresenter.swift:135]: `array1` and `array2` are bad names.

**Comment by Ivan-Skoryk** [comment]
@Artem-Kalmykov check pls

**Pull request merged**